@here/harp-datasource-protocol/lib/StyleSetEvaluator.ts

```diff
@@ -355,6 +358,8 @@ export class StyleSetEvaluator {
 
         const searchedStyleSet = this.getOptimizedStyleSet(optimizedSubSetKey);
 
+        const previousZoomLevel = this.changeZoomLevel(env.lookup("$zoom") as number | undefined);
```

Comment:
This executes `env.lookup` for each feature, and most calls to `getMatchingTechniques` from OMVDecoder already know actual zoomLevel.
I would add `zoomLevel` to argumentlist of `getMatchingTechniques` ...

Reply:
`$zoom` is something that we compute based the tile storage level and the storage level offsets, currently we compute $zoom in `createFeatureEnv()` and pass the Env to the the visit methods. So we need to refactor that code in order to remove the lookup for $zoom.

Reply:
Ok, will be in followup PR, which will set zoom at start of decoding of tile.